vertical-pod-autoscaler/hack/deploy-for-e2e-locally.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
+
+function print_help {
+  echo "ERROR! Usage: deploy-for-e2e-locally.sh [suite]*"
+  echo "<suite> should be one of:"
+  echo " - recommender"
+  echo " - recommender-externalmetrics"
+  echo "If component is not specified all above will be started."
+}
+
+if [ $# -eq 0 ]; then
+  print_help
+  exit 1
+fi
+
+if [ $# -gt 1 ]; then
+  print_help
+  exit 1
+fi
+
+SUITE=$1
+
+case ${SUITE} in
+  recommender|recommender-externalmetrics)
+    COMPONENTS="${SUITE}"
+    ;;
+  *)
+    print_help
+    exit 1
+    ;;
+esac
+
+# Local KIND-hosted registry
+export REGISTRY=localhost:5001
+export TAG=latest
+
+rm -f ${SCRIPT_ROOT}/hack/e2e/vpa-rbac.yaml
+patch -c ${SCRIPT_ROOT}/deploy/vpa-rbac.yaml -i ${SCRIPT_ROOT}/hack/e2e/vpa-rbac.diff -o ${SCRIPT_ROOT}/hack/e2e/vpa-rbac.yaml
+kubectl apply -f ${SCRIPT_ROOT}/hack/e2e/vpa-rbac.yaml
+# Other-versioned CRDs are irrelevant as we're running a modern-ish cluster.
+kubectl apply -f ${SCRIPT_ROOT}/deploy/vpa-v1-crd-gen.yaml
+kubectl apply -f ${SCRIPT_ROOT}/hack/e2e/k8s-metrics-server.yaml
+
+for i in ${COMPONENTS}; do
+  if [ $i == admission-controller ] ; then
```

Comment:
If I understand correctly this should never happen. Check in line 43 ensures that `COMPONENTS` is `recommender` or `recommender-externalmetrics`.